draft-ietf-quic-transport.md

```diff
@@ -322,175 +322,356 @@ all field sizes are in bits.  When discussing individual bits of fields, the
+
+: The remaining 5 bits of octet 0 include one of 32 packet types.
+  {{short-packet-types}} lists the types that are defined for short packets.
+
+Connection ID:
+
+: If the Connection ID Flag is set, a connection ID occupies octets 1 through 8
+  of the packet.  See {{connection-id}} for more details.
+
+Packet Number:
+
+: The length of the packet number field depends on the packet type.  This field
+  can be 1, 2 or 4 octets long depending on the short packet type.
+
+Encrypted Payload:
+
+: Packets with a short header always include a 1-RTT protected payload.
+
+The packet type in a short header currently determines only the size of the
+packet number field.  Additional types can be used to signal the presence of
+other fields.
+
+| Type | Packet Number Size     |
+|:-----|:-----------------------|
+| 01   | 1 octet                |
+| 02   | 2 octets               |
+| 03   | 4 octets               |
+{: #short-packet-types title="Short Header Packet Types"}
+
+The header form, connection ID flag and connection ID of a short header packet
+are version independent.  The remaining fields are specific to the selected QUIC
+version.  See {{version-specific}} for details on how packets from different
+versions of QUIC are interpreted.
 
-~~~
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                        [Version (32)]                         |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                  Packet Number (8/16/32/48)                 ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                    {Encrypted Payload (*)}                  ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-~~~
-{: #regular-packet-format title="Regular Packet"}
 
-The fields in a Regular packet past the Common Header are the following:
+## Version Negotiation Packet {#version-negotiation-packet}
 
-* QUIC Version: A 32-bit opaque tag that represents the version of the QUIC
-  protocol.  Only present in the client-to-server direction, and if the VERSION
-  flag is set.  Version Negotiation is described in {{version-negotiation}}.
+A Version Negotiation packet is sent by only the server and is a response to a
+client packet of an unsupported version. It uses a long header and contains:
 
-* Packet Number: The lower 8, 16, 32, or 48 bits of the packet number, based on
-  the PACKET_NUMBER_SIZE flag.  Each Regular packet is assigned a packet number
-  by the sender.  The first packet number is randomized (see
-  {{initial-packet-number}}).
+* Octet 0: 0x81 (Flags indicating long header and Version Negotiation packet
+  type)
+* Octets 1-8: Connection ID (echoed)
+* Octets 9-12: Packet Number (echoed)
+* Octets 13-16: Version (echoed)
+* Octets 17+: Payload (version list, containing 0 or more acceptable versions)
```

Comment:
Yeah, it seems a little odd to allow the server to respond "I speak just enough QUIC to say 'I don't speak QUIC.'"